zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -173,6 +183,25 @@ public static Either<ProblemDetail, UpdateJobRequest> toJobUpdateRequest(
                 getLongOrZero(updateRequest, r -> updateRequest.getChangeset().getTimeout())));
   }
 
+  public static Either<ProblemDetail, DocumentCreateRequest> toDocumentCreateRequest(
+      final String documentId, final String storeId, final MultipartFile file,
+      final DocumentMetadata metadata) {
+    final var validationResponse = validateDocumentMetadata(metadata);
+    final var internalMetadata = toInternalDocumentMetadata(metadata, file);
+    final InputStream inputStream;
+    try {
+      inputStream = file.getInputStream();
+    } catch (IOException e) {
+      // should be handled globally as an internal server error
+      throw new RuntimeException("Failed to read document content", e);
```

Comment:
🔧 I would recommend handling it here explicitly instead of having it handled globally. You could either
1. Create another violation and collect it together with the ones from `validationResponse` by returning a list of violations there instead of a `ProblemDetail`.
2. Handle this with precedence and return an HTTP 500 `ProblemDetail` that "overwrites" the `validationResponse`.